overlord/snapstate/snapmgr.go

```diff
@@ -224,3 +228,25 @@ func (m *SnapManager) Wait() {
 func (m *SnapManager) Stop() {
 	m.runner.Stop()
 }
+
+// SnapInfo returns snap.Info for a given snap name and snap version.
+//
+// Snap name and version are used to locate the snap.yaml in the filesystem.
+// Some information is not present in the YAML and is looked up in the state
+// instead.
```

Comment:
```
// SnapInfo returns the snap.Info for a snap in the system.
//
// Today this function is looking at data directly from the mounted snap, but soon it will
// be changed so it looks first at the state for the snap details (Revision, Developer, etc),
// and then complements it with information from the snap itself.
```